azurestack/resource_arm_loadbalancer_nat_pool.go

```diff
@@ -0,0 +1,287 @@
+			d.Set("frontend_ip_configuration_id", feipConfig.ID)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmLoadBalancerNatPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).loadBalancerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	loadBalancerID := d.Get("loadbalancer_id").(string)
+	armMutexKV.Lock(loadBalancerID)
+	defer armMutexKV.Unlock(loadBalancerID)
+
+	loadBalancer, exists, err := retrieveLoadBalancerById(loadBalancerID, meta)
+	if err != nil {
+		return fmt.Errorf("Error retrieving LoadBalancer by ID: %+v", err)
+	}
+	if !exists {
+		d.SetId("")
+		return nil
+	}
+
+	_, index, exists := findLoadBalancerNatPoolByName(loadBalancer, d.Get("name").(string))
+	if !exists {
+		return nil
+	}
+
+	oldNatPools := *loadBalancer.LoadBalancerPropertiesFormat.InboundNatPools
+	newNatPools := append(oldNatPools[:index], oldNatPools[index+1:]...)
+	loadBalancer.LoadBalancerPropertiesFormat.InboundNatPools = &newNatPools
+
+	resGroup, loadBalancerName, err := resourceGroupAndLBNameFromId(d.Get("loadbalancer_id").(string))
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer Name and Group: {{err}}", err)
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, loadBalancerName, *loadBalancer)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Load Balancer %q (Resource Group %q): %+v", loadBalancerName, resGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for completion of the Load Balancer %q (Resource Group %q): %+v", loadBalancerName, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, loadBalancerName, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Load Balancer: %+v", err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read LoadBalancer %q (Resource Group %q) ID", loadBalancerName, resGroup)
+	}
+
+	return nil
+}
+
+func expandAzureRmLoadBalancerNatPool(d *schema.ResourceData, lb *network.LoadBalancer) (*network.InboundNatPool, error) {
+
+	properties := network.InboundNatPoolPropertiesFormat{
+		Protocol:               network.TransportProtocol(d.Get("protocol").(string)),
+		FrontendPortRangeStart: utils.Int32(int32(d.Get("frontend_port_start").(int))),
+		FrontendPortRangeEnd:   utils.Int32(int32(d.Get("frontend_port_end").(int))),
+		BackendPort:            utils.Int32(int32(d.Get("backend_port").(int))),
+	}
+
+	if v := d.Get("frontend_ip_configuration_name").(string); v != "" {
+		rule, _, exists := findLoadBalancerFrontEndIpConfigurationByName(lb, v)
+		if !exists {
+			return nil, fmt.Errorf("[ERROR] Cannot find FrontEnd IP Configuration with the name %s", v)
+		}
+
+		feip := network.SubResource{
+			ID: rule.ID,
+		}
+
+		properties.FrontendIPConfiguration = &feip
+	}
+
+	natPool := network.InboundNatPool{
+		Name: utils.String(d.Get("name").(string)),
+		InboundNatPoolPropertiesFormat: &properties,
+	}
+
+	return &natPool, nil
```

Comment:
this could become

```go
return &network.InboundNatPool{
		Name: utils.String(d.Get("name").(string)),
		InboundNatPoolPropertiesFormat: &properties,
	}, nil
```